### ISSUE TITLE
[VPAT][A11y] disabled password Textfield is missing role in voice over

### ISSUE THREAD

**Issue by chunhtai** [flutter/flutter#173049]
b/429089551

Run the https://github.com/flutter/flutter/tree/master/dev/a11y_assessments and navigate to textfield password label and focus disabled password field

<img width="1035" height="1872" alt="Image" src="https://github.com/user-attachments/assets/74710ef4-4394-44b7-815a-724297aca17d" />

VoiceOver announces: "Password, disabled".

EXPECTED RESULTS:

VoiceOver announces "Password, Text field, disabled" or similar.

**Comment by victorsanni** [comment]
This is happening because disabled text field is [internally classified as read-only](https://github.com/flutter/flutter/blob/fbbc797a6a802f265eca423679ef95c1c02845b7/packages/flutter/lib/src/material/text_field.dart#L1677). And read-only text field has a [non-text-input semantics object](https://github.com/flutter/flutter/blob/d328555151ca0a2e5455c3cc9900b43dc4867d54/engine/src/flutter/shell/platform/darwin/ios/framework/Source/accessibility_bridge.mm#L267) because selectable text is also internally classified as read-only.

The native iOS text field with VoiceOver announces "Password, dimmed, text field".

**Comment by chunhtai** [comment]
This is tricky, the reason that we don't create text input semantics node is because a read-only textfield is used by selectable text. we need a better indication for selectabletext instead of inferring from read-only textfield
